akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ClientSpec.scala

```diff
@@ -274,6 +279,31 @@ class Http2ClientSpec extends AkkaSpecWithMaterializer("""
 
       }
     }
+
+    "support stream support for receiveing response entity data" should {
+      abstract class WaitingForResponseSetup extends TestSetup with NetProbes with Http2FrameHpackSupport {
+        val streamId = 0x1
+        emitRequest(streamId, Get("/"))
+        expectDecodedHEADERS(streamId, endStream = true)
+      }
+      "support trailing headers for responses" in new WaitingForResponseSetup {
+        sendHEADERS(streamId, endStream = false, Seq(
+          RawHeader(":status", "200"),
+          RawHeader("content-type", "application/octet-stream")
+        ))
+
+        val response = expectResponse()
+        response.entity shouldBe a[Chunked]
+
+        sendDATA(streamId, endStream = false, ByteString("asdf"))
+        sendHEADERS(streamId, endStream = true, Seq(RawHeader("grpc-status", "0")))
+
+        val chunks = response.entity.asInstanceOf[Chunked].chunks.runWith(Sink.seq).futureValue
+        chunks(0) should be(Chunk("asdf"))
+        chunks(1) should be(LastChunk(extension = "", List(RawHeader("grpc-status", "0"))))
+      }
```

Comment:
👍 

Perhaps add coverage for failing if headers are incorrectly emitted mid stream as well.

Reply:
BTW failing on headers mid stream has a ticket #3268